cfgov/v1/models/handlers.py

```diff
@@ -0,0 +1,78 @@
+from collections import OrderedDict
+from itertools import chain
+from wagtail.wagtailcore import blocks
+from wagtail.wagtailcore.blocks.stream_block import StreamValue
+
+from ..atomic_elements import ATOMIC_JS
+
+
+class Handler(object):
+    def __init__(self, page, request):
+        self.page = page
+        self.request = request
+
+    def process(self, context):
+        raise NotImplementedError
+
+    # Retrieves the stream values on a page from it's Streamfield
+    def get_streamfield_blocks(self):
+        lst = [value for key, value in vars(self.page).iteritems()
+               if type(value) is StreamValue]
+        return list(chain(*lst))
+
+
+class JSHandler(Handler):
+    """
+        Gathers all the JS files specific to this page and its current
+        Streamfield's blocks and puts them in the template context.
+    """
+    def __init__(self, page, request):
+        super(JSHandler, self).__init__(page, request)
+        self.js_dict = OrderedDict()
+
+    def process(self, context):
+        self.generate_js_dict()
+        context['media'] = self.js_dict
```

Comment:
@chosak This is true, I'll make a change that checks for the existence of `media`. There's a future PR that actually uses Wagtail's hooks module to register these handlers instead.